src/main/java/org/reactome/addlinks/fileprocessors/XenbaseFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBioMartUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class XenbaseFileProcessor extends FileProcessor{
+
+    public XenbaseFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public XenbaseFileProcessor()
+    {
+        super();
+    }
+
+    private static final int uniprotIdentifierIndex = 0;
+    private static final int xenbaseIdentifierIndex = 3;
+
+    /**
+     * Build map of UniProt identifiers to Xenbase identifiers that is used to create Xenbase cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFilePath = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+
+        List<String> lines = new ArrayList<>();
+        try {
+            lines = EnsemblBioMartUtil.getLinesFromFile(inputFilePath, false);
+        } catch (IOException e) {
+            logger.error("Error reading file ({}): {}", inputFilePath.toString(), e.getMessage());
+            e.printStackTrace();
+        }
```

Comment:
It might be nice to move the try/catch blocks into the `EnsemblBioMartUtil.getLinesFromFile` method itself and have it return an empty list when an exception is thrown.  That way the main code can call it directly:

`for (String line : EnsemblBioMartUtil.getLinesFromFile(inputFilePath, false)) {`

Reply:
Ya good catch, I made the try/catch block specific to the location where the Exception can come from, but then I turned that location into its own method